plugins/train/trainer/_base.py

```diff
@@ -176,10 +175,11 @@ def train_one_step(self, viewer, timelapse_kwargs):
             for side, batcher in self.batchers.items():
                 if self.pingpong.active and side != self.pingpong.side:
                     continue
-                loss[side] = batcher.train_one_batch(do_preview)
+                loss[side] = batcher.train_one_batch()
                 if not do_preview and not do_timelapse:
                     continue
                 if do_preview:
+                    batcher.generate_preview(do_preview)
```

Comment:
train_one_batch should have a single code function... to train one batch. Not also to create a preview sample if the iteration needs a preview. Moving code to separate distinct functions